avro-serializer/src/main/java/io/confluent/kafka/serializers/KafkaAvroDeserializer.java

```diff
@@ -39,18 +39,23 @@ public KafkaAvroDeserializer(SchemaRegistryClient client) {
 
   public KafkaAvroDeserializer(SchemaRegistryClient client, Map<String, ?> props) {
     schemaRegistry = client;
-    configure(deserializerConfig(props));
+    configure(deserializerConfig(props), false);
+  }
+
+  public KafkaAvroDeserializer(SchemaRegistryClient client, Map<String, ?> props, boolean isKey) {
+    schemaRegistry = client;
+    configure(deserializerConfig(props), isKey);
   }
 
   @Override
   public void configure(Map<String, ?> configs, boolean isKey) {
     this.isKey = isKey;
-    configure(new KafkaAvroDeserializerConfig(configs));
+    configure(new KafkaAvroDeserializerConfig(configs), isKey);
```

Comment:
As mentioned above, this should use `isKey` and figure out what `Class<?> type` to pass to `configure`